src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/IndirectionNode.cs

```diff
@@ -12,19 +12,40 @@ namespace ILCompiler.DependencyAnalysis
     /// </summary>
     public class IndirectionNode : ObjectNode, ISymbolNode
     {
-        private ISymbolNode _indirectedNode;
+        public class IndirectionNodeData
+        {
+            public ISymbolNode IndirectedNode { get; private set; }
+            public int OffsetDelta { get; private set; }
+            public IndirectionNodeData(ISymbolNode indirectedNode, int offsetDelta = 0)
+            {
+                IndirectedNode = indirectedNode;
+                OffsetDelta = offsetDelta;
+            }
+            public override bool Equals(object obj)
+            {
+                IndirectionNodeData other = obj as IndirectionNodeData;
+                return other != null && other.IndirectedNode == IndirectedNode && other.OffsetDelta == OffsetDelta;
+            }
+            public override int GetHashCode()
+            {
+                return IndirectedNode.GetHashCode();
+            }
+        }
+
+        private IndirectionNodeData _indirectionData;
         private TargetDetails _target;
 
-        public IndirectionNode(TargetDetails target, ISymbolNode indirectedNode)
+        public IndirectionNode(TargetDetails target, IndirectionNodeData indirectionData)
```

Comment:
Cc @hoyMS. This is the change I'm talking about in another review. Please make sure this doesn't cause unnecessary merge conflicts with your changes.